compiler/next/include/chpl/uast/ForwardingDecl.h

```diff
@@ -0,0 +1,126 @@
+#ifndef CHPL_UAST_FORWARDING_H
+#define CHPL_UAST_FORWARDING_H
+
+#include "chpl/queries/Location.h"
+#include "chpl/uast/VisibilityClause.h"
+#include "chpl/uast/Expression.h"
+#include "chpl/uast/Decl.h"
+#include "chpl/uast/Variable.h"
+
+namespace chpl {
+namespace uast {
+
+
+/**
+  This class represents a forwarding statement. Forwarding allows a record
+  or class to specify that certain method calls will be forwarded to a
+  particular expression. For example:
+
+  \rst
```

Comment:
IMO this example is a bit too much for the compiler docs. (It's fine for the technote or language spec or primer but the compiler docs have a different purpose). I'd recommend we instead here focus on two examples - one `forwarding var` and one that is not - and talk about the uAST that represents them.